### DISCUSSION TITLE
All screen's Providers are shown at dropdown selection, although component is not under them at screen edit at backend.

### DISCUSSION THREAD

**Discussion by gmentzik** [Budibase/budibase#13055] [Help] [answered]
Hi. 
We are testing version 2.19.5.

We have seen the following behavior. When selecting a provider for a picker all page providers are displayed instead of just the providers the component is directly under at the hierarchy tree.
Checked with picker and table.

Please check captures below.

![image](https://github.com/Budibase/budibase/assets/65217733/62fc68a1-8dbe-497c-b611-210720568a37)

![image](https://github.com/Budibase/budibase/assets/65217733/a115ddcb-3438-4224-9ffb-c833997df26a)

![image](https://github.com/Budibase/budibase/assets/65217733/bbf82cde-3fa7-46a5-abd1-2a184fd7d0fc)

Good thing is that selection of invalid provider is not accepted, but it is confusing.

Please let us know if you can reproduce the issue!

**Comment by kasgithub8** [comment]
@gmentzik wait is this an issue? or is it just this update [#12926](https://github.com/Budibase/budibase/discussions/12926)?

**Comment by ConorWebb96** [comment, marked as answer]
Hey @gmentzik,

This is likely related to the [global bindings](https://github.com/Budibase/budibase/discussions/12926) changes released recently. I believe i see what you are saying, any non nested provides don't have bindable labels or values if you select them. Only the one that has been nested will allow you to see/select the fields?

I would recommend that you rename your data provider components so the names are less confusing.

I believe this to be a bug, I've written this up [here](https://github.com/Budibase/budibase/issues/13065).

**Comment by gmentzik** [comment]
> Hey @gmentzik,
> 
> This is likely related to the [global bindings](https://github.com/Budibase/budibase/discussions/12926) changes released recently. I believe i see what you are saying, any non nested provides don't have bindable labels or values if you select them. Only the one that has been nested will allow you to see/select the fields?
> 
> I would recommend that you rename your data provider components so the names are less confusing.
> 
> I believe this to be a bug, I've written this up [here](https://github.com/Budibase/budibase/issues/13065).

Hi @ConorWebb96 ,

Yes only nested have bindable labels and values.  I find this new feature useful but a bit confusing on Provider rules. Before with nesting it was clear to which data you had access. Now you can access any provider, (after the bug is resolved). This has the positive that you can reuse   a data source (less queries) but I have a small fear it may introduce confusion. Hope it goes well.
